### PR TITLE
Regression test for #417

### PR DESCRIPTION
**Pull request by mdickinson**
This PR adds a regression test for the issue described in #417 (where extension point contributions were being resolved before plugins had started).

The new test should fail on master, but pass once #417 is solved.

### CODE REVIEW

**Comment by mdickinson** [comment]
We got the expected failure on Travis:
```
======================================================================
FAIL: test_extension_point_resolution_occurs_after_plugin_start (envisage.tests.test_application.ApplicationTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/.edm/envs/envisage-test-3.6-null/lib/python3.6/site-packages/envisage/tests/test_application.py", line 329, in test_extension_point_resolution_occurs_after_plugin_start
    [1, 2, 3],
AssertionError: Lists differ: [] != [1, 2, 3]
Second list contains 3 additional elements.
First extra element 0:
1
- []
+ [1, 2, 3]
```

**Review by aaronayres35 — APPROVED** [review]
LGTM once #422 is merged

**Comment by rahulporuri** [comment]
> LGTM once #422 is merged

#422 has been merged and I updated this PR, which is now green. Looks like #422 removed all offensive code. We can now merge this PR.

**Pull request merged**
